weather_mv/loader_pipeline/pipeline.py

```diff
@@ -41,12 +44,75 @@ def pattern_to_uris(match_pattern: str) -> t.Iterable[str]:
         yield from [x.path for x in match.metadata_list]
 
 
+def validate_region(output_table: str, temp_location: t.Optional[str] = None, region: t.Optional[str] = None) -> None:
+    """Validates non-compatible regions scenarios by performing sanity check."""
+
+    def do_cleanup(bigquery_client: bigquery.Client, storage_client: storage.Client, dummy_output_table: str,
+                   dummy_bucket_name: str) -> None:
+        bigquery_client.delete_table(dummy_output_table, not_found_ok=True)
+        try:
+            storage_client.get_bucket(dummy_bucket_name).delete(force=True)
+        except NotFound:
+            pass
```

Comment:
I like the pattern you have here. And, I'm glad that you're cleaning up resources. I think that a context manager would be a great form factor to perform this work, instead of a try..except.finally block. This seems to be a better fit if you intend to create a separate function, which I agree is a bit cleaner. 

https://docs.python.org/3/library/contextlib.html

Reply:
I see now that how this is called. If cleanup is required before the test, then a context manager is probably not a good idea. In this case, I think it'd still be better to make this a private function at the top level.

Reply:
Done.